src/mpc-hc/mplayerc.cpp

```diff
@@ -347,207 +349,212 @@ CStringA GetContentType(CString fn, CAtlList<CString>* redir)
             ishttp = true;
         }
 
-        if (!ishttp && _tcsicmp(url.GetSchemeName(), _T("https")) != 0) {
+        if (_tcsicmp(url.GetSchemeName(), _T("https")) == 0) {
+            ishttp = true;
+        }
+
+        if (!ishttp) {
             return "";
         }
     }
-
+    // Get content type by getting the header response from server
     if (ishttp) {
-        DWORD ProxyEnable = 0;
-        CString ProxyServer;
-        DWORD ProxyPort = 0;
-        ULONG len = 256 + 1;
-        CRegKey key;
-
-        if (ERROR_SUCCESS == key.Open(HKEY_CURRENT_USER, _T("Software\\Microsoft\\Windows\\CurrentVersion\\Internet Settings"), KEY_READ)
-                && ERROR_SUCCESS == key.QueryDWORDValue(_T("ProxyEnable"), ProxyEnable) && ProxyEnable
-                && ERROR_SUCCESS == key.QueryStringValue(_T("ProxyServer"), ProxyServer.GetBufferSetLength(256), &len)) {
-            ProxyServer.ReleaseBufferSetLength(len);
-
-            CAtlList<CString> sl;
-            ProxyServer = Explode(ProxyServer, sl, ';');
-            if (sl.GetCount() > 1) {
-                POSITION pos = sl.GetHeadPosition();
-                while (pos) {
-                    CAtlList<CString> sl2;
-                    if (!Explode(sl.GetNext(pos), sl2, '=', 2).CompareNoCase(_T("http"))
-                            && sl2.GetCount() == 2) {
-                        ProxyServer = sl2.GetTail();
-                        break;
-                    }
-                }
-            }
-
-            ProxyServer = Explode(ProxyServer, sl, ':');
-            if (sl.GetCount() > 1) {
-                ProxyPort = _tcstol(sl.GetTail(), nullptr, 10);
-            }
+        CInternetSession internet;
+
+#pragma warning(push)
+#pragma warning(disable: 4996)
+        OSVERSIONINFOEX osVersion = { sizeof(OSVERSIONINFOEX) };
+        GetVersionEx(reinterpret_cast<LPOSVERSIONINFO>(&osVersion));
+#pragma warning(pop)
+        CString osVersionStr;
+        osVersionStr.Format(_T("Windows %1u.%1u"), osVersion.dwMajorVersion, osVersion.dwMinorVersion);
+
+#if !defined(_WIN64)

```

Comment:
Use just "MPC-HC" as user agent.

Reply:
Done. While coding it, I was wondering about implementing a configurable global variable for the User-Agent. Some servers might only respond to web browsers...

Reply:
@clsid2: What about an advanced options "Use browser User-Agent" with true/false that when true use an User-Agent similar to the one of youtube-dl?
This: `Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3768.2 Safari/537.36`

To note: they identify as Windows 10 also on the versions of Windows.